File: class2Arrays/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string userText = get_string("Text: ");
    int numL = count_letters(userText);
    int numW = count_words(userText);
    int numS = count_sentences(userText);
    
    float L =   (float) numL / (float) numW * 100;
    float S =   (float) numS / (float) numW * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);  
    
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
    
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters ++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1; //+ 1 for the last word in text 
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 32) // 32 Ascii for space
        {
            words ++;
        }
    }
    return words;
}
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63) // 33 ! 46 . 63 ?
        {
            sentences ++;
        }
    }
    return sentences;
}

File: class6Python/readability/readability.py
'''readabilty.py - prompts user for text and outputs its grade level'''

from cs50 import get_string

def main():
    text = get_string("Text: ")
    
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100
    
    index = round(0.0588 * L - 0.296 * S - 15.8) 

    if index > 15:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")

def count_letters(text):
    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters

def count_words(text):
    # + 1 for the last word in text
    words = 1
    for i in range(len(text)):
        if text[i] == " ":
            words += 1
    return words

def count_sentences(text):
    sentences = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "!" or text[i] == "?":
            sentences += 1
    return sentences

if __name__ == "__main__":
    main()